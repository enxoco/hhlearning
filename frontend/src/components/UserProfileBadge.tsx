import { Box, HStack, Text } from '@chakra-ui/react'

type UserProfileProps = {
  name: string
  email: string
}

export default function UserProfileBadge({ name, email }: UserProfileProps) {
  const initials = [...new Set(name.replace(/ /g, "").split(/[a-z]/))].join("");
  return (
    <HStack spacing="3" ps="2">
      <div aria-label={name} role="img" className="initials-avatar"><div>{initials}</div></div>
      <Box data-testid="userProfileBadge">
        <Text fontWeight="medium" fontSize="sm" role='paragraph'>
          {name}
        </Text>
        <Text color="muted" fontSize="sm" role="paragraph">
          {email}
        </Text>
      </Box>
    </HStack>
  )
}