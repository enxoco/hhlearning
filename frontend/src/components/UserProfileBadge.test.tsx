import { render } from "@testing-library/react"
import '@testing-library/jest-dom';

import UserFactory from "#/factories/users";
import UserProfileBadge from "./UserProfileBadge";

const Users = new UserFactory();

test("profile badge renders correctly", () => {
    const users = Users.CreateUsers(3);
    expect(users).toHaveLength(3);
    const component = render(<UserProfileBadge name={users[0].name} email={users[0].email}  />);
    
    for (let i = 0; i < users.length; i++) {
        component.rerender(<UserProfileBadge name={users[i].name} email={users[i].email}  />)
        // our component should render 2 p tags.
        // 1 with full name and 1 with email
        const pTags = component.getAllByRole("paragraph");
        expect(pTags).toHaveLength(2)
        expect(pTags[0]).toHaveTextContent(users[i].name)
        expect(pTags[1]).toHaveTextContent(users[i].email)
    }
})