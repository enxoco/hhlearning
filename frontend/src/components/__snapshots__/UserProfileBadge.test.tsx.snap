// Vitest Snapshot v1

exports[`profile badge renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="chakra-stack css-1dq2vho"
      >
        <div
          aria-label="Marquis Gutmann"
          class="initials-avatar"
          role="img"
        >
          <div>
            MG
          </div>
        </div>
        <div
          class="css-0"
          data-testid="userProfileBadge"
        >
          <p
            class="chakra-text css-w75v0a"
            role="paragraph"
          >
            Marquis Gutmann
          </p>
          <p
            class="chakra-text css-1s07rjm"
            role="paragraph"
          >
            Marquis_Gutmann69@yahoo.com
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-stack css-1dq2vho"
    >
      <div
        aria-label="Marquis Gutmann"
        class="initials-avatar"
        role="img"
      >
        <div>
          MG
        </div>
      </div>
      <div
        class="css-0"
        data-testid="userProfileBadge"
      >
        <p
          class="chakra-text css-w75v0a"
          role="paragraph"
        >
          Marquis Gutmann
        </p>
        <p
          class="chakra-text css-1s07rjm"
          role="paragraph"
        >
          Marquis_Gutmann69@yahoo.com
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
